geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionHolder.java

```diff
@@ -791,4 +791,19 @@ public boolean equals(Object obj) {
     return canon;
   }
 
+  private synchronized boolean isRegionSynchronizeScheduledOrDone() {
+    return regionSynchronizeScheduledOrDone;
+  }
+
+  public synchronized void setRegionSynchronizeScheduled() {
+    regionSynchronizeScheduledOrDone = true;
+  }
+
+  public synchronized boolean setRegionSynchronizeScheduledOrDoneIfNot() {
+    if (!isRegionSynchronizeScheduledOrDone()) {
+      regionSynchronizeScheduledOrDone = true;
+      return true;
+    }
```

Comment:
if isRegionSynchronizeScheduledOrDone == true, do you need to set it to false before returning false?

Reply:
No. This will possibly cause multiple regions sync message being sent by the same node.

If a bucket region has more than 2 redundant copies, or replicate regions has more than 2 nodes. Each node will start a timed task to synchronize with all existing hosts for the region. That means each node would receive more than 2 region sync messages. By setting the flag to true and keep it to true, it will ensure only one region sync message being sent by one node.